bindings/rust/wasmedge-sys/src/instance/function.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{wasmedge, Value, WasmFnIO, HOST_FUNCS};
+
+use core::ffi::c_void;
+use std::convert::TryInto;
+
+use rand::Rng;
+
+extern "C" fn wraper_fn(
+    key_ptr: *mut c_void,
+    _data: *mut c_void,
+    _mem_cxt: *mut wasmedge::WasmEdge_MemoryInstanceContext,
+    params: *const wasmedge::WasmEdge_Value,
+    param_len: u32,
+    returns: *mut wasmedge::WasmEdge_Value,
+    return_len: u32,
+) -> wasmedge::WasmEdge_Result {
+    let key = key_ptr as *const usize as usize;
+    let mut result: Result<Vec<Value>, u8> = Err(0);
+
+    let mut input: Vec<Value> = {
+        let raw_input = unsafe {
+            std::slice::from_raw_parts(
+                params,
+                param_len
+                    .try_into()
+                    .expect("len of params should not greater than usize"),
+            )
+        };
+        raw_input.iter().map(|r| (*r).into()).collect()
+    };
+    input.remove(0);
+
+    let return_len = return_len
+        .try_into()
+        .expect("len of returns should not greater than usize");
+    let raw_returns = unsafe { std::slice::from_raw_parts_mut(returns, return_len) };
+
+    HOST_FUNCS.with(|f| {
+        let mut host_functions = f.borrow_mut();
+        let real_fn = host_functions
+            .remove(&key)
+            .expect("host function should there");
```

Comment:
`should be there` or something? I have no idea about this error message.

Reply:
Before the VM starting, the host function registry and the key return, we now use the key to get the host function back. So the host function is expected in the Hashmap.